sklearn/pipeline.py

```diff
@@ -664,12 +685,32 @@ def _transform_one(transformer, X, y, weight, **fit_params):
     return res * weight
 
 
-def _fit_transform_one(transformer, X, y, weight, **fit_params):
-    if hasattr(transformer, 'fit_transform'):
-        res = transformer.fit_transform(X, y, **fit_params)
-    else:
-        res = transformer.fit(X, y, **fit_params).transform(X)
-    # if we have a weight for this transformer, multiply output
+def _fit_transform_one(transformer,
+                       X,
+                       y,
+                       weight,
+                       return_transform=False,
```

Comment:
This feels quite strange: a `_fit_transform_` method that just uses `fit` depending on the `return_transform` argument. Might be worth considering 2 different methods